biodatasets/citation_gia_test_collection/citation_gia_test_collection.py

```diff
@@ -0,0 +1,342 @@
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir[0]+"\GNormPlusCorpus", "BC2GNtest.BioC.xml"),
+                    "split": "train",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir[0]+"\GNormPlusCorpus", "BC2GNtrain.BioC.xml"),
+                    "split": "test",
+                },
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir[0]+"\GNormPlusCorpus", "NLMIAT.BioC.xml"),
+                    "split": "NLMIAT",
+                },
+            ),
+        ]
+
+    def _get_entities(self, annot_d: dict) -> dict:
+        ''''
+        Converts annotation dict to entity dict.
+        '''
+        ent = {
+            "id": str(uuid.uuid4()),
+            "type": annot_d["type"],
+            "text": [annot_d["text"]],
+            "offsets": [annot_d["offsets"]],
+            "normalized": [
+                {
+                    "db_name": "NCBI Gene" if annot_d["type"].isdigit() else "",
+                    "db_id": annot_d["type"] if annot_d["type"].isdigit() else "",
+                }
+            ],
+        }
+
+        return ent
+
+    def _get_offsets_entities(child, parent_text: str, child_text: str, offset: int) -> List[int]:
+        '''
+        Extracts child text offsets from parent text for entities. 
+        Some offsets that were present in the datset were wrong mainly because of string encodings.
+        Also a little fraction of parent strings doesn't contain its respective child strings. 
+        Hence few assertion errors in the entitity offsets checking test. 
+        '''
+        if child_text in parent_text:
+            index = parent_text.index(child_text)
+            start = index + offset
+
+        else:
+            start = offset
+        end = start + len(child_text)
+
+        return [start, end]
+
+    def _process_annot(self, annot: ET.Element, passages: dict) -> dict:
+        ''''
+        Converts annotation XML Element to Python dict.
+        '''
+        parent_text = " ".join([p['text'] for p in passages.values()])
+        annot_d = dict()
+        a_d = {a.tag: a.text for a in annot}
+
+        for a in list(annot):
+
+            if a.tag == "location":
+                offset = int(a.attrib["offset"])
+                annot_d["offsets"] = self._get_offsets_entities(
+                    html.escape(parent_text[offset:]),
+                    html.escape(a_d["text"]), offset)
+
+            elif a.tag != "infon":
+                annot_d[a.tag] = html.escape(a.text)
+
+            else:
+                annot_d[a.attrib["key"]] = a_d["infon"]
+
+        return annot_d
+
+    def _parse_elem(self, elem: ET.Element) -> dict:
```

Comment:
I am not asking you to rewrite the script but to load BioC file you should really consider using the [bioc library](https://github.com/bionlplab/bioc)

Reply:
I didn't really know such library exists! Thank you!